packages/ui5-task-zipper/lib/zipper.js

```diff
@@ -0,0 +1,46 @@
+
+const yazl = require("yazl");
+const resourceFactory = require("@ui5/fs").resourceFactory;
+const log = require("@ui5/logger").getLogger("builder:customtask:zipper");
+
+module.exports = async function ({ workspace, dependencies, options }) {
+
+    const zipName = `${options.configuration.archiveName || options.projectNamespace.replace(/\//g, '')}.zip`;
+    const prefixPath = `/resources/${options.projectNamespace}/`
+    let allResources;
+    try {
+        allResources = await workspace.byGlob(`${prefixPath}/**`);
+    } catch (e) {
+        log.error(`Couldn't read resources: ${e}`);
+    }
+
+    const zip = new yazl.ZipFile();
+    try {
+        await Promise.all(allResources.map((resource) =>
+            resource.getBuffer()
+                .then((buffer) => {
+                    options.configuration && options.configuration.debug && log.info(`Adding ${resource.getPath()} to archive.`);
+                    zip.addBuffer(buffer, resource.getPath().replace(prefixPath, ''));
+                })
+        ));
+    } catch (e) {
+        log.error(`Couldn't add all resources to the archive: ${e}`);
```

Comment:
closing the ZIP should be OK - but it might make no sense to continue from here and to create an incomplete zip, right? Maybe you throw an error again here to interrupt the build or to silently fail with the error message but then you also should not create the zip file.

Reply:
That makes sense, yes